### PR TITLE
[21286] Improve CI workflows in master

### PR DESCRIPTION
**Pull request by JesusPoderoso**
## Description

This PR improves the CI workflows, making them as similar as possible to the Fast DDS workflows.
It also introduces a braching model refactor, similiar as the performed in Fast DDS repository.

It is on top of (and must be merged after) this PR:
- #203 

**Note**: While backporting this PR, please ensure the following changes:
- Remove nightly jobs
- Update `ubuntu-ci.yml`, `mac-ci.yml` and `windows-ci.yml` files with the proper fallback branches
- Maintain reduced version of the RELEASE_SUPPORT.md (only reference to master links)
- Create mirror job on 1.1.x brach (targetting 1.x)
- Fast CDR 1.0.x requires building on `ubuntu-20.04`

@Mergifyio backport 2.2.x 2.1.x 1.1.x 1.0.x 

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- **N/A** The code follows the style guidelines of this project. 
- **N/A** Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- **N/A** Any new/modified methods have been properly documented using Doxygen. 
- **N/A** Changes are backport compatible: they do **NOT** break ABI nor change library core behavior. 
- **N/A** Changes are API compatible. 
- **N/A** New feature has been added to the `versions.md` file (if applicable).
- [x] Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [ ] The title and description correctly express the PR's purpose.
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: CI pass and failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by JesusPoderoso** [comment]
Note: I am not sure if the correct milestone should be v3.0.0 or v2.3.0

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/403/)
* Mac Build Status
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/385/)

File: .github/workflows/reusable-ubuntu-ci.yml
Comment (EduPonz) [inline .github/workflows/reusable-ubuntu-ci.yml:1]: I think we can make do with only one "multiplatform" reusable. The only thing missing is the toolset, but I think we could have it as an opt-in arg that only takes effect on windows

File: RELEASE_SUPPORT.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Release support
+
+
+Please, refer to the [main branch](https://github.com/eProsima/Fast-CDR/blob/master/RELEASE_SUPPORT.md) for the latest version of this document.
```
Comment (EduPonz) [inline RELEASE_SUPPORT.md:4]:
```suggestion
Please, refer to the [master branch](https://github.com/eProsima/Fast-CDR/blob/master/RELEASE_SUPPORT.md) for the latest version of this document.
```

File: RELEASE_SUPPORT.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+*eProsima Fast CDR* maintains several releases with different support cycles.
+**All of them are attached to different *eProsima Fast DDS* releases.**
+
+## *eProsima Fast DDS* and *Fast CDR* version compatibility
+
+|Fast DDS Version|Fast DDS Python Version|Fast DDS Python Version branch|Fast DDS Python Latest Release|
```
Comment (EduPonz) [inline RELEASE_SUPPORT.md:11]:
```suggestion
|Fast DDS Version|Fast CDR Version|Fast CDR Version branch|Fast CDR Latest Release|
```

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/405/)
* Mac Build Status
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/387/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/406/)
* Mac Build Status
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/388/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/407/)
* Mac Build Status
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/389/)

**Comment by JesusPoderoso** [comment]
Closed in favor of:
- https://github.com/eProsima/Fast-CDR/pull/216

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/410/)
* Mac Build Status
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/392/)

**Pull request closed without merging**